src/k8s/pkg/k8sd/api/worker.go

```diff
@@ -36,7 +37,10 @@ func (e *Endpoints) postWorkerInfo(s state.State, r *http.Request) response.Resp
 		return response.InternalError(fmt.Errorf("failed to get cluster config: %w", err))
 	}
 
-	certificates := pki.NewControlPlanePKI(pki.ControlPlanePKIOpts{Years: 10})
+	// NOTE: Default certificate expiration is set to 10 years.
+	defaultExpiration := int(time.Now().AddDate(10, 0, 0).Sub(time.Now()) / time.Second)
```

Comment:
nit: maybe this is slightly more readable: 
```
int(time.Until(time.Now().AddDate(10, 0, 0)) / time.Second)
```

Reply:
this will definitely pile up to off-by-one or a few seconds issues, see my other comment about how to deal with this